javascript/ql/src/semmle/javascript/frameworks/Logging.qll

```diff
@@ -62,7 +62,40 @@ private module Console {
       then result = getArgument([1 .. getNumArgument()])
       else result = getAnArgument()
     }
+
+    /**
+     * Gets the name of the console logging method, e.g. "log", "error", "assert", etc.
+     */
+    string getName() {
+      result = name
+    }
   }
+
+ /**
+  * A call to a function that forwards all arguments to some console logging mechanism.
+  * The called function contain a single statement similar to: `console.log.apply(this, arguments)`.
+  */
+ class IndirectConsoleLoggerCall extends LoggerCall {
```

Comment:
What if we add the following more general changes instead?

- add `getASpreadArgument()` to `ConsoleLoggerCall::getAMessageComponent`
- separately implement a taint arg/param step from a tainted argument to the entire `arguments` array.

Wouldn't that be a more general solution?

Reply:
I've changed the implementation such that the `arguments` object is tainted. now.

It is an additional taint step on the specific query. If we like it we can add it as a general taint-step later.